jmclient/jmclient/wallet_service.py

```diff
@@ -536,6 +539,42 @@ def display_rescan_message_and_system_exit(self, restart_cb):
                 jmprint(restart_msg, "important")
                 sys.exit(EXIT_SUCCESS)
 
+    def sync_burner_outputs(self, burner_txes):
+        mixdepth = FidelityBondMixin.FIDELITY_BOND_MIXDEPTH
+        address_type = FidelityBondMixin.BIP32_BURN_ID
+        self.wallet.set_next_index(mixdepth, address_type, self.wallet.gap_limit,
```

Comment:
Having read this whole function, it makes sense to me, except I don't understand why the index is set to the gap limit and then the loop is only over that set of values. Does it mean you only have 6 burn addresses available?

Reply:
In practice people will only ever have one burn output at most. But yes you're right limiting to 6 pubkeys is subpar, I'll try to recode it to somehow loop until 6 unused pubkeys are found.

Reply:
I studied this a bit more.

That code should only be run when the user first synchronizes their burner outputs. As you know burner outputs must be sync'd by scanning all wallet transactions, so it can only be done in `-recoversync` mode. Once the burn transactions are found they are added to the `wallet.jmdat` file and will be available normally. So users should increase the gap limit with `-g` when they're first sync'ing after they've made a burner transaction.

So no change is needed.